### PR TITLE
Docker image

### PR DESCRIPTION
**Pull request by ChrisTerBeke**
Add support to build an Docker image with the entire build environment. Inspired by our own Jenkins CentOS setup, but then updated.

### CODE REVIEW

**Comment by ChrisTerBeke** [comment]
There's something wrong with Jenkins, it just doesn't build (not a build error).

File: README.md [hunk 1]
```diff
@@ -123,3 +123,13 @@ nmake
...
 the build systems of the sub-projects that are built.
+
+
+### Docker
+Building the environment in Docker is very easy using the Dockerfile in this repository.
+We handle all cloning and compiling in the Dockerfile, so no dependencies are needed on the host OS.
+The resulting image can be used as a `FROM` for building Cura itself.
+
+```bash
+docker build -t ultimaker/cura-build-environment:1 .
```
Comment (LipuFei) [inline README.md]: I think the tag `1` is not necessary here.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    zlib-devel
+
+# Enable devtools-3
+ENV PATH=${PATH}:/opt/rh/devtoolset-3/root/usr/bin
+
+# Init xstartup
+ADD ./xstartup /
+RUN mkdir /.vnc
+RUN x11vnc -storepasswd 123456 /.vnc/passwd
+RUN  \cp -f ./xstartup /.vnc/.
```
Comment (LipuFei) [inline Dockerfile]: This `\cp` seems to be a mistake?

Comment (ChrisTerBeke) [reply Dockerfile]: mm indeed

File: README.md (hunk 1 above)
Comment (ChrisTerBeke) [reply README.md]: It's a specific version of the image (now 1).

File: Dockerfile (hunk 2 above)
Comment (ChrisTerBeke) [reply Dockerfile]: Fixed in latest commit.

File: README.md (hunk 1 above)
Comment (ChrisTerBeke) [reply README.md]: Fixed in latest commit.

**Comment by ChrisTerBeke** [comment]
@LipuFei I've changed some things, can you re-review?

**Pull request merged**